include/task_manager.h

```diff
@@ -88,14 +89,22 @@ namespace detail {
 		 */
 		void shutdown() { task_map.clear(); }
 
+		void set_horizon_step(const int step) { task_horizon_step_size = step; }
```

Comment:
I assume negative step sizes are semantically invalid, so this should probably have an `assert(step >= 0)` to document and sanity check the provided value.

Reply:
Agreed, in fact I think it must even be `> 1` (right?).

Reply:
I don't think it necessarily has to be. In fact, one of the graph compaction tests tests step size 1.